RiotSwiftUI/Modules/UserSessions/UserOtherSessions/UserOtherSessionsViewModel.swift

```diff
@@ -21,72 +21,98 @@ typealias UserOtherSessionsViewModelType = StateStoreViewModel<UserOtherSessions
-        super.init(initialViewState: UserOtherSessionsViewState(bindings: UserOtherSessionsBindings(filter: filter),
+        defaultTitle = title
+        let bindings = UserOtherSessionsBindings(filter: filter, isEditModeEnabled: false)
+        let items = filter.filterSessionInfos(sessionInfos: sessionInfos, selectedSessions: selectedSessions)
+        super.init(initialViewState: UserOtherSessionsViewState(bindings: bindings,
                                                                 title: title,
-                                                                sections: []))
-        updateViewState()
+                                                                items: items,
+                                                                header: filter.userOtherSessionsViewHeader,
+                                                                emptyItemsTitle: filter.userOtherSessionsViewEmptyResultsTitle,
+                                                                allItemsSelected: false))
     }
     
     // MARK: - Public
     
     override func process(viewAction: UserOtherSessionsViewAction) {
         switch viewAction {
         case let .userOtherSessionSelected(sessionId: sessionId):
-            guard let session = sessionInfos.first(where: { $0.id == sessionId }) else {
-                assertionFailure("Session should exist in the array.")
-                return
+            if state.bindings.isEditModeEnabled {
+                updateSelectionForSession(sessionId: sessionId)
+                updateViewState()
+            } else {
+                showUserSessionOverview(sessionId: sessionId)
             }
-            completion?(.showUserSessionOverview(sessionInfo: session))
         case .filterWasChanged:
             updateViewState()
         case .clearFilter:
             state.bindings.filter = .all
             updateViewState()
+        case .editModeWasToggled:
+            selectedSessions.removeAll()
+            updateViewState()
+        case .toggleAllSelection:
+            toggleAllSelection()
+            updateViewState()
         }
     }
-    
+
     // MARK: - Private
     
-    private func updateViewState() {
-        let sectionItems = createSectionItems(sessionInfos: sessionInfos, filter: state.bindings.filter)
-        let sectionHeader = createHeaderData(filter: state.bindings.filter)
-        if sectionItems.isEmpty {
-            state.sections = [.emptySessionItems(header: sectionHeader,
-                                                 title: noSessionsTitle(filter: state.bindings.filter))]
-        } else {
-            state.sections = [.sessionItems(header: sectionHeader,
-                                            items: sectionItems)]
+    private func showUserSessionOverview(sessionId: String) {
+        guard let session = sessionInfos.first(where: { $0.id == sessionId }) else {
+            assertionFailure("Session should exist in the array.")
```

Comment:
You can use `MXLog.failure` instead, which will both log the issue into rageshakes and raise `assertionFailure` if in `DEBUG` build

Reply:
Great, thanks

Reply:
Wait, I think I can't use `MXLog` because this file is part of RiotSwifUI target